File: scratch/test_wp.py
import urllib.request
import urllib.parse
import json

def test():
    slug = "sonic-x-shadow-generations-ps-plus-extra-vale-download"
    url = f"https://dougdesign.com.br/wp-json/wp/v2/posts?slug={slug}&_embed=true"
    print(f"Fetching: {url}")
    
    req = urllib.request.Request(
        url,
        headers={"User-Agent": "Mozilla/5.0"}
    )
    try:
        with urllib.request.urlopen(req, timeout=10) as response:
            data = json.loads(response.read().decode("utf-8"))
            if not data:
                print("No post found.")
                return
            
            post = data[0]
            print(f"Title: {post['title']['rendered']}")
            print(f"Featured Media ID: {post['featured_media']}")
            
            embedded = post.get("_embedded", {})
            media_list = embedded.get("wp:featuredmedia", [])
            if media_list:
                media = media_list[0]
                print(f"Source URL: {media['source_url']}")
            else:
                print("No embedded featured media found.")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    test()

File: scratch/test_wp.js
async function test() {
  const slug = "sonic-x-shadow-generations-ps-plus-extra-vale-download";
  const url = `https://dougdesign.com.br/wp-json/wp/v2/posts?slug=${slug}&_embed=true`;
  console.log(`Fetching: ${url}`);
  try {
    const res = await fetch(url);
    const data = await res.json();
    if (!data || data.length === 0) {
      console.log("No post found.");
      return;
    }
    const post = data[0];
    console.log(`Title: ${post.title.rendered}`);
    console.log(`Featured Media ID: ${post.featured_media}`);
    const mediaList = post._embedded?.["wp:featuredmedia"];
    if (mediaList && mediaList.length > 0) {
      console.log(`Source URL: ${mediaList[0].source_url}`);
    } else {
      console.log("No embedded featured media found.");
    }
  } catch (e) {
    console.error("Error:", e);
  }
}
test();
